packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -241,25 +217,52 @@ export class OperationsFactory {
                 resolveTree,
                 context,
                 operation,
+                whereArgs: resolveTreeWhere,
             });
         } else {
-            const concreteConnectionOperations = target.concreteEntities.map(
-                (concreteEntity: ConcreteEntityAdapter) => {
+            let concreteConnectionOperations: ConnectionReadOperation[] = [];
+            if (isInterfaceEntity(target)) {
+                const nodeWhere: Record<string, any> = isObject(resolveTreeWhere) ? resolveTreeWhere.node : {};
+                const concreteEntities = filterConcreteEntities(target, nodeWhere);
+                concreteConnectionOperations = concreteEntities.map((concreteEntity: ConcreteEntityAdapter) => {
                     const connectionPartial = new InterfaceConnectionPartial({
                         relationship,
                         directed,
                         target: concreteEntity,
                     });
+                    // nodeWhere with the shared filters applied
+                    const concreteNodeWhere = getConcreteWhere(target, concreteEntity, nodeWhere);
+                    const whereArgs = { edge: resolveTreeWhere.edge ?? {}, node: concreteNodeWhere };
 
                     return this.hydrateConnectionOperationAST({
                         relationship,
                         target: concreteEntity,
                         resolveTree,
                         context,
                         operation: connectionPartial,
+                        whereArgs: whereArgs,
                     });
-                }
```

Comment:
These big blocks are getting complex to follow, please move them to separate functions if possible